pkg/manifest/objects_test.go

```diff
@@ -0,0 +1,249 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package manifest
+
+import "testing"
+
+func TestHash(t *testing.T) {
+	hashTests := []struct {
+		kind      string
+		namespace string
+		name      string
+		want      string
+	}{
+		{"aaa", "nnn", "mmm", "aaa:nnn:mmm"},
+		{"a_", "n_", "m_", "a_:n_:m_"},
+		{"a:a", "n:n", "m:m", "a:a:n:n:m:m"},
+		{"", "", "", "::"},
+	}
+
+	for _, tt := range hashTests {
+		got := Hash(tt.kind, tt.namespace, tt.name)
+		if got != tt.want {
+			t.Errorf("got %s for kind %s, namespace %s, name %s, want %s", got, tt.kind, tt.namespace, tt.name, tt.want)
+		}
+	}
+}
+
+func TestHashNameKind(t *testing.T) {
+	hashNameKindTests := []struct {
+		kind string
+		name string
+		want string
+	}{
+		{"aaa", "nnn", "aaa:nnn"},
+		{"a_", "n_", "a_:n_"},
+		{"a:a", "n:n", "a:a:n:n"},
+		{"", "", ":"},
+	}
+
+	for _, tt := range hashNameKindTests {
+		got := HashNameKind(tt.kind, tt.name)
+		if got != tt.want {
+			t.Errorf("got %s for kind %s, name %s, want %s", got, tt.kind, tt.name, tt.want)
+		}
+	}
+}
+
+func TestParseJSONToK8sObject(t *testing.T) {
+	testDeployment := `{
```

Comment:
would be easier to read and more compact to express these in yaml.

Reply:
have added test cases for both parse from json and yaml.